src/docfx/lib/github/GitHubAccessor.cs

```diff
@@ -42,39 +43,85 @@ public async Task<(Error, GitHubUser)> GetUserByLogin(string login)
                 return (_rateLimitError, null);
             }
 
-            var apiDetail = $"GET /users/{login}";
+            if (_unauthorizedError != null)
+            {
+                return default;
+            }
+
+            var query = @"
+query ($login: String!) {
+  user(login: $login) {
+    name
+    email
+    databaseId
+  }
+}";
+
             try
             {
-                var user = await RetryUtility.Retry(
-                    () => _client.User.Get(login),
-                    ex => ex is OperationCanceledException);
+                var (error, response) = await Query(query, new
+                {
+                    login,
+                });
+
+                if (error != null)
+                {
+                    return (error, null);
+                }
+
+                var grahpApiResponse = JsonConvert.DeserializeAnonymousType(await response.Content.ReadAsStringAsync(), new
+                {
+                    errors = new[]
+                    {
+                        new
+                        {
+                            type = "",
+                            path = Array.Empty<string>(),
+                            message = "",
+                        },
+                    },
+                    data = new
+                    {
+                        user = new
+                        {
+                            name = "",
+                            email = "",
+                            login = "",
+                            databaseId = 0,
+                        },
+                    },
+                });
+
+                if (grahpApiResponse.errors != null && grahpApiResponse.errors.Length != 0)
+                {
+                    var notFoundError = grahpApiResponse.errors.FirstOrDefault(e => e.type == "NOT_FOUND" && e.path.Contains("user"));
+                    if (notFoundError != null)
+                        return default;
+
+                    var rateLimitError = grahpApiResponse.errors.FirstOrDefault(e => e.type == "MAX_NODE_LIMIT_EXCEEDED" || e.type == "RATE_LIMITED");
+                    if (rateLimitError != null)
+                    {
+                        _rateLimitError = Errors.GitHubApiFailed(_url, rateLimitError.message);
+                    }
+
+                    return (Errors.GitHubApiFailed(_url, rateLimitError?.message ?? grahpApiResponse.errors.First().message), null);
```

Comment:
The github api failed error message is already very long, consider remove the detailed diagnostic message from error to hide it from error report, write it using `Log.Write` instead.

Reply:
I would like expose this error message to end user to let them know the reason.

Reply:
the message is really helpful.

![image](https://user-images.githubusercontent.com/19990166/58461635-b5e56a80-8162-11e9-842c-8cc6ef85149e.png)

Reply:
I once see a full json dumped into the message, it it because line 286 simply dumps the whole JSON response?

Reply:
this part is fine, only including the message

Reply:
unless we give the wrong query, we should get clear and clean error message